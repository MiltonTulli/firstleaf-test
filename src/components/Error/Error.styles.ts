import { makeStyles } from '@material-ui/core/styles';
import { Theme } from '@material-ui/core';

export default makeStyles<Theme>(theme => ({
  root: {
    width: '100%',
    height: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  paper: {
    padding: theme.spacing(2),
    minWidth: 300
  },
  message: {
    marginTop: theme.spacing(1)
  }
}));
