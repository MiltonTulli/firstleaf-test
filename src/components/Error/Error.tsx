import React from 'react';
import { Typography, Paper, Divider } from '@material-ui/core';
import styles from './Error.styles';

interface IErrorProps {
  children: React.ReactNode;
  render?: () => JSX.Element;
  message: string;
  hasError: boolean;
  title?: string;
}
export const Error = (props: IErrorProps) => {
  const classes = styles();
  const { children, hasError, title = 'Error', message, render } = props;
  return hasError ? (
    <div className={classes.root}>
      <Paper className={classes.paper}>
        <Typography data-testid="title" color="error" variant="h4">
          {title}
        </Typography>
        <Divider />
        <Typography className={classes.message} color="error" data-testid="message" variant="body2">
          {message}
        </Typography>
        {render && <div>{render()}</div>}
      </Paper>
    </div>
  ) : (
    <> {children} </>
  );
};
