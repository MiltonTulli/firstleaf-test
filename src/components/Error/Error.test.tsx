import React from 'react';
import { shallow } from 'enzyme';
import { Error } from './Error';

describe('<Error>', () => {
  let props;
  beforeEach(() => {
    props = {
      render: jest.fn(() => <h1 data-testid="render">hello</h1>),
      message: 'MESSAGE',
      hasError: true,
      title: 'ERROR_TITLE'
    };
  });
  it('Should match snapshot', () => {
    const wrapper = shallow(<Error {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should show default title', () => {
    delete props.title;
    const wrapper = shallow(<Error {...props} />);
    const node = wrapper.find('[data-testid="title"]');
    expect(node.text()).toEqual('Error');
  });

  it('Should show custom title', () => {
    const wrapper = shallow(<Error {...props} />);
    const node = wrapper.find('[data-testid="title"]');
    expect(node.text()).toEqual(props.title);
  });

  it('Should show message', () => {
    const wrapper = shallow(<Error {...props} />);
    const node = wrapper.find('[data-testid="message"]');
    expect(node.text()).toEqual(props.message);
  });

  it('Should render prop', () => {
    const wrapper = shallow(<Error {...props} />);
    const node = wrapper.find('[data-testid="render"]');
    expect(props.render).toHaveBeenCalled();
    expect(node).toBeTruthy();
  });

  it('Should not render prop', () => {
    props.render = null;
    const wrapper = shallow(<Error {...props} />);
    const node = wrapper.find('[data-testid="render"]');
    expect(node.exists()).toBeFalsy();
  });

  it('Should not render children if has error', () => {
    props.hasError = true;
    const wrapper = shallow(
      <Error {...props}>
        <div data-testid="child">child</div>
      </Error>
    );
    const node = wrapper.find('[data-testid="child"]');
    expect(node.exists()).toBeFalsy();
  });

  it('Should render children if does not has error', () => {
    props.hasError = false;
    const wrapper = shallow(
      <Error {...props}>
        <div data-testid="child">child</div>
      </Error>
    );
    const node = wrapper.find('[data-testid="child"]');
    expect(node.exists()).toBeTruthy();
  });
});
