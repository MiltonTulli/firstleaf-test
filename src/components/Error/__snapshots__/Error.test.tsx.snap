// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error> Should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-2"
  >
    <WithStyles(ForwardRef(Typography))
      color="error"
      data-testid="title"
      variant="h3"
    >
      ERROR_TITLE
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-message-3"
      color="error"
      data-testid="message"
      variant="body2"
    >
      MESSAGE
    </WithStyles(ForwardRef(Typography))>
    <div>
      <h1
        data-testid="render"
      >
        hello
      </h1>
    </div>
  </WithStyles(ForwardRef(Paper))>
</div>
`;
